pkg/apply/apply.go

```diff
@@ -24,16 +24,22 @@ type Object interface {
 	runtime.Object
 }
 
+// GetClusterName returns the names.ClusterNameAnnotation annotation for the specified object.
+// If the annotation does not exist it will return an empty string.
+func GetClusterName(obj Object) string {
+	return obj.GetAnnotations()[names.ClusterNameAnnotation]
```

Comment:
This should probably return the default cluster name if empty, since it's useful in other circumstances.

Reply:
Hmm, now I'm not so sure.

Reply:
We rely on empty cluster name if we mean `default` I think we can keep this as is

Reply:
agreed